sklearn/model_selection/split.py

```diff
@@ -0,0 +1,1106 @@
+
+            train = rng.permutation(train)
+            test = rng.permutation(test)
+
+            yield train, test
+
+    def __repr__(self):
+        return ('%s(n_iter=%d, test_size=%s, random_state=%s)' % (
+                    self.__class__.__name__,
+                    self.y,
+                    self.n_iter,
+                    str(self.test_size),
+                    self.random_state,
+                ))
+
+    def __len__(self):
+        return self.n_iter
+
+
+class PredefinedSplit(_PartitionIterator):
+    """Predefined split cross validation iterator
+
+    Splits the data into training/test set folds according to a predefined
+    scheme. Each sample can be assigned to at most one test set fold, as
+    specified by the user through the ``test_fold`` parameter.
+
+    Read more in the :ref:`User Guide <split>`.
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import PredefinedSplit
+    >>> X = np.array([[1, 2], [3, 4], [1, 2], [3, 4]])
+    >>> y = np.array([0, 0, 1, 1])
+    >>> ps = PredefinedSplit()
+    >>> print(ps)       # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+    sklearn.model_selection.split.PredefinedSplit()
+    >>> test_folds = [0, 1, -1, 1]
+    >>> for train_index, test_index in ps.split(X, y, labels=test_folds)
+    ...    print("TRAIN:", train_index, "TEST:", test_index)
+    ...    X_train, X_test = X[train_index], X[test_index]
+    ...    y_train, y_test = y[train_index], y[test_index]
+    TRAIN: [1 2 3] TEST: [0]
+    TRAIN: [0 2] TEST: [1 3]
+    """
+    def _pre_split_check(self, X, y=None, labels=None):
+        return
+
+    def _iter_test_indices(self, X, y=None, labels=None):
+        test_fold = column_or_1d(np.array(labels, dtype=np.int))
+        unique_folds = np.unique(test_fold)
+        unique_folds = unique_folds[unique_folds != -1]
+        for f in unique_folds:
+            yield np.where(test_fold == f)[0]
+
+    def __repr__(self):
+        return '%s.%s()' % (
+            self.__class__.__module__,
+            self.__class__.__name__)
+
+
+def check_cv(cv, y=None, classifier=False):
+    """Input checker utility for building a CV in a user friendly way.
+
+    Parameters
+    ----------
+    cv : int, a cv generator instance, or None
+        The input specifying which cv generator to use. It can be an
+        integer, in which case it is the number of folds in a KFold,
+        None, in which case 3 fold is used, or another object, that
+        will then be used as a cv generator.
+
+    y : array-like
+        The target variable for a supervised learning problem.
+
+    classifier : boolean optional
+        Whether the task is a classification task, in which case
+        stratified KFold will be used.
+
+    Returns
+    -------
+    checked_cv: a cross-validation generator instance.
+        The return value is guaranteed to be a cv generator instance, whatever
+        the input type.
+    """
+    if cv is None:
```

Comment:
Since we need to support `None`, I left it as such... however having the default as 3, will make it easier for people to know that from the function signature! Should I change it?